crates/sql-backend-handler/src/sql_opaque_handler.rs

```diff
@@ -60,12 +60,38 @@ impl SqlBackendHandler {
             .await?
             .and_then(|u| u.0))
     }
+
+    #[instrument(skip(self), level = "debug", err)]
+    async fn check_user_enabled(&self, user_id: &UserId) -> Result<bool> {
```

Comment:
Let's fold this into `get_password_file_for_user`, that way:
- we reduce the number of round-trips to the DB by querying all the info at once
- We make sure everyone who reads the password file also respects the login_enabled bool.

Reply:
Although I see that you're also using it for the OPAQUE login, in the finish part. In that case, you can keep this function, but let's _also_ make get_password_file query both fields.